src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs

```diff
@@ -141,8 +141,15 @@ public void LoadProxyBypassList()
 
             foreach (string bypass in proxyBypassEnv.Split(new [] {',', ';'}).Where(value => !string.IsNullOrWhiteSpace(value)).Select(value => value.Trim()))
             {
-                Trace.Info($"Bypass proxy for: {bypass}.");
-                ProxyBypassList.Add(bypass);
+                string saveRegexString = bypass;
+
+                var regExp = new Regex("(?<!\\\\)([.])(?!\\*)");
+                var replace = "\\$1";
+
+                saveRegexString = regExp.Replace(saveRegexString, replace);
```

Comment:
Could it make sense to extract this logic into a separate method and add documentation regarding why we are processing the list of proxies in that way?

Reply:
done